src/videojs-hls.js

```diff
@@ -331,7 +331,7 @@ videojs.Hls.prototype.addCuesForMetadata_ = function(segmentInfo) {
       frame = metadata.frames[i];
       time = segmentOffset + ((metadata.pts - minPts) * 0.001);
       cue = new window.VTTCue(time, time, frame.value || frame.url || '');
```

Comment:
You can detect whether DataCue or VTTCue is available and use the appropriate one where available:
```js
new (DataCue || WebKitDataCue || VTTCue)(0, 1, 'hi this is text');
```